rclcpp/include/rclcpp/subscription.hpp

```diff
@@ -150,24 +150,25 @@ class Subscription : public SubscriptionBase
         context,
         this->get_topic_name(),    // important to get like this, as it has the fully-qualified name
         qos.get_rmw_qos_profile(),
-        resolve_intra_process_buffer_type(options.intra_process_buffer_type, callback)
+        resolve_intra_process_buffer_type(options.intra_process_buffer_type, callback),
+        get_subscription_handle().get()
         );
 
       // Add it to the intra process manager.
       using rclcpp::experimental::IntraProcessManager;
       auto ipm = context->get_sub_context<IntraProcessManager>();
       uint64_t intra_process_subscription_id = ipm->add_subscription(subscription_intra_process);
       this->setup_intra_process(intra_process_subscription_id, ipm);
+    } else {
+      TRACEPOINT(
+        rclcpp_subscription_callback_added,
+        (const void *)get_subscription_handle().get(),
+        (const void *)&any_callback_);
+      // The callback object gets copied, so if registration is done too early/before this point
+      // (e.g. in `AnySubscriptionCallback::set()`), its address won't match any address used later
+      // in subsequent tracepoints.
+      any_callback_.register_callback_for_tracing();
     }
-
-    TRACEPOINT(
-      rclcpp_subscription_callback_added,
-      (const void *)get_subscription_handle().get(),
-      (const void *)&any_callback_);
-    // The callback object gets copied, so if registration is done too early/before this point
-    // (e.g. in `AnySubscriptionCallback::set()`), its address won't match any address used later
-    // in subsequent tracepoints.
-    any_callback_.register_callback_for_tracing();
```

Comment:
@christophebedard I'm not entirely sure you want to remove this bit. Bear in mind that `Subscription` and `SubscriptionIntraProcess` instances keep separate copies of `AnyCallback` and dispatch to each separately. Don't you want to trace both calls?

Reply:
@hidmic it was my understanding that `Subscription`'s copy of `AnySubscriptionCallback` is _never_ used if intra-process is enabled. Thus, in this case, we only need to worry about/register `SubscriptionIntraProcess`'s `AnySubscriptionCallback`'s copy. Is this correct? Or can both be used?

Reply:
`Subscription`'s copy of `AnySubscriptionCallback` is used for inter-process communication, see [here](https://github.com/ros2/rclcpp/blob/b92db52bb14ec097946bfe44d2b337359dddfbab/rclcpp/include/rclcpp/subscription.hpp#L210-L220).

Reply:
I see. In this case, having them use the same `AnySubscriptionCallback` object would really be nice, but like we discussed this is the better solution for now. Fixed with 416423f675efef362eb4ff3971c07469a9a57afb!